experimental/dds/tree2/src/feature-libraries/typed-schema/buildViewSchemaCollection.ts

```diff
@@ -229,3 +225,50 @@ export function validateField(
 	// 	);
 	// }
 }
+
+function validateStructFieldName(
+	name: string,
+	describeField: () => string,
+	errors: string[],
+): void {
+	const bannedNames = new Set([
```

Comment:
It might be prudent to add a comment next to UntypedTree that this function exists, so they are less likely to get out of sync. And/or have the module that contains UntypedTree also export a list of names used, and then this function can use that list of names. Then at least two of the things that might get out of sync are in the same module.

Reply:
We have a test that will fail if this is ever out of sync with the implementation, though I forgot to update it when I moved this here. I'll fix that.

Reply:
Fixed.